core/Config/IniFileChain.php

```diff
@@ -210,13 +210,16 @@ public function reload($defaultSettingsFiles = array(), $userSettingsFile = null
             $this->resetSettingsChain($defaultSettingsFiles, $userSettingsFile);
         }
 
-        if (!empty($userSettingsFile) && !empty($GLOBALS['ENABLE_CONFIG_PHP_CACHE'])) {
+        $hasAbsoluteConfigFile = !empty($userSettingsFile) && strpos($userSettingsFile, DIRECTORY_SEPARATOR) === 0;
+        $useConfigCache = !empty($GLOBALS['ENABLE_CONFIG_PHP_CACHE']) && $hasAbsoluteConfigFile;
+
+        if ($useConfigCache) {
             $cache = new Cache();
             $values = $cache->doFetch(self::CONFIG_CACHE_KEY);
             
             if (!empty($values)
                 && isset($values['mergedSettings'])
-                && isset($values['settingsChain'])) {
+                && isset($values['settingsChain'][$userSettingsFile])) {
```

Comment:
Here we make sure the cached entry actually has `$values['settingsChain']['/home/path/config/config.ini.php']` and is not eg `$values['settingsChain']['../../config.ini.php']` . It shouldn't be needed probably but it protects using the cache maybe in some other edge cases where the app is sending a different absolute path than expected for some reason. Eg should there be a double slash somewhere in the path etc.